presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry.py

```diff
@@ -126,67 +93,17 @@ def load_predefined_recognizers(
         :param nlp_engine: The NLP engine to use.
         :return: None
         """
-        if not languages:
-            languages = ["en"]
-
-        recognizers_map = {
-            "en": [
-                UsBankRecognizer,
-                UsLicenseRecognizer,
-                UsItinRecognizer,
-                UsPassportRecognizer,
-                UsSsnRecognizer,
-                NhsRecognizer,
-                SgFinRecognizer,
-                AuAbnRecognizer,
-                AuAcnRecognizer,
-                AuTfnRecognizer,
-                AuMedicareRecognizer,
-                InPanRecognizer,
-                InAadhaarRecognizer,
-                InVehicleRegistrationRecognizer,
-                InVoterRecognizer,
-                InPassportRecognizer,
-            ],
-            "es": [
-                EsNifRecognizer,
-                EsNieRecognizer,
-            ],
-            "it": [
-                ItDriverLicenseRecognizer,
-                ItFiscalCodeRecognizer,
-                ItVatCodeRecognizer,
-                ItIdentityCardRecognizer,
-                ItPassportRecognizer,
-            ],
-            "pl": [PlPeselRecognizer],
-            "ALL": [
-                CreditCardRecognizer,
-                CryptoRecognizer,
-                DateRecognizer,
-                EmailRecognizer,
-                IbanRecognizer,
-                IpRecognizer,
-                MedicalLicenseRecognizer,
-                PhoneRecognizer,
-                UrlRecognizer,
-            ],
-        }
-        for lang in languages:
-            lang_recognizers = [
-                self.__instantiate_recognizer(
-                    recognizer_class=rc, supported_language=lang
-                )
-                for rc in recognizers_map.get(lang, [])
-            ]
-            self.recognizers.extend(lang_recognizers)
-            all_recognizers = [
-                self.__instantiate_recognizer(
-                    recognizer_class=rc, supported_language=lang
-                )
-                for rc in recognizers_map.get("ALL", [])
-            ]
-            self.recognizers.extend(all_recognizers)
+
+        from .recognizer_registry_provider import RecognizerRegistryProvider
+        registry_configuration = {"global_regex_flags": self.global_regex_flags}
+        if languages is not None:
+            registry_configuration["supported_languages"] = languages
+
+        recognizers = RecognizerRegistryProvider(
```

Comment:
There are two issues with a new module:
1. `recognizer_registry_provider` was made to load recognizers from a config file, so I'm not sure we should extract it any further, it's pretty thin in functionality. 
2. We can't create this logic in the registry because we have no nlp engine at this point (which could be scpaCy or transformer or whatever. 

The only entity that is aware of both the registry and the nlp engine is the `analyzer_engine`. How about we create this logic here? and just make sure to keep on using the same logic?